### PR TITLE
Remove vulnerabilities from the service & update go version to 1.18

### PR DESCRIPTION
**Pull request by PauloPortugal**
### What
* We are not using `github.com/gorilla/sessions` and there is a non-CVE
vulnerability found.

* Get the latest versions
*  Update the Go version to 1.18

### How to review

* sense check
* Ensure tests pass

### Who can review

Any ONS developer

### CODE REVIEW

File: ci/build.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.17
+    tag: 1.8.3
```
Comment (rafahop) [inline ci/build.yml:9]: 1.18.3 - Also should we update component.yml?

Comment (PauloPortugal) [reply ci/build.yml:9]: I have addressed the comment :) Thank you :)

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,16 @@
 module github.com/ONSdigital/dp-population-types-api
 
-go 1.17
-
-replace github.com/coreos/etcd => github.com/coreos/etcd v3.3.24+incompatible
+go 1.18
 
 replace github.com/dgrijalva/jwt-go v3.2.0+incompatible => github.com/golang-jwt/jwt v3.2.2+incompatible
```
Comment (rafahop) [inline go.mod:5]: Is this replace still required?

Comment (PauloPortugal) [reply go.mod]: I have addressed your comment :)

**Review by rafahop — APPROVED** [review]

**Comment by PauloPortugal** [comment]
@rafahop could you review this again please?
I had to rebase develop into my branch.
Thank you

**Review by rafahop — APPROVED** [review]

**Pull request merged**